mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt

```diff
@@ -0,0 +1,146 @@
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.mifos.mobilewallet.model.domain.BankAccountDetails
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.bank.viewmodel.BankAccountsViewModel
+
+@Composable
+fun AccountScreen(viewModel: BankAccountsViewModel,
+                  onAddAccountClicked: () -> Unit,
+                  onAccountClicked: (BankAccountDetails) -> Unit){
+    val bankAccountDetails = viewModel.bankAccountDetailsList
+    AccountDisplay(bankAccountDetails,onAddAccountClicked,onAccountClicked)
+}
+
+@Composable
+fun AccountDisplay(
+    bankAccountDetails: List<BankAccountDetails>,
+    onAddAccountClicked: () -> Unit,
+    onAccountClicked: (BankAccountDetails) -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+    ) {
+        Text(
+            text = stringResource(id = R.string.linked_bank_account),
+            fontSize = 16.sp,
+            color = colorResource(id = R.color.colorTextPrimary),
+            modifier = Modifier.padding(
+                top = 48.dp,
+                start = 24.dp )
+        )
+        if (bankAccountDetails.isEmpty()) {
+            Column(
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxWidth()
+                    .padding(top = 48.dp, bottom = 70.dp) ,
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+
+            ){
+            PlaceholderStateLayout(
+                title =stringResource(id = R.string.empty_no_accounts_title),
+                subtitle = stringResource(id =R.string.empty_no_accounts_subtitle),
+                drawableResId = R.drawable.ic_empty_state
+            )
+            }
+        }
+        else {
+                LazyColumn(
+                    modifier = Modifier
+                        .weight(1f)
+                        .padding(top = 48.dp)
+                        .fillMaxSize(),
+                    state = rememberLazyListState()
+                ) {
+                    items(bankAccountDetails.size) { index ->
+                        val bankAccount = bankAccountDetails[index]
+                                    ItemCausualList(
+                                        bankAccount = bankAccount,
+                                        onAccountClicked
+                                    )
+                    }
+                }
+        }
+        Spacer(modifier = Modifier.height(16.dp))
+        AddAccountChip(
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .padding(bottom = 128.dp),
+            onAddAccountClicked = onAddAccountClicked
+        )
+    }
+}
+
+@Preview
+@Composable
+fun AccountScreenPreview()  {
```

Comment:
Empty bank details preview is missing

Reply:
created the new preview function for empty list